### DISCUSSION TITLE
swc plugin to mimic babel plugin-debug-label plugin

### DISCUSSION THREAD

**Discussion by axelson** [pmndrs/jotai#1057] [Q&A]
Is there any way to use https://github.com/pmndrs/jotai/blob/b466ed954bf7f90698f4a8dcce06dbc4d0ac8709/src/babel/plugin-debug-label.ts with swc instead of babel?

Here's the swc plugin guide:
https://swc.rs/docs/usage/plugins

But it seems like a re-implementation of the plugin will be required.

**Comment by dai-shi** [comment]
Yeah, I guess @Thisen is interested in it.

**Reply by Thisen** [reply]
Already working on it :)

**Reply by notsidney** [reply]
If it helps, I’ve found that sticking with Babel for the dev server and using swc for production works :)

**Reply by Thisen** [reply]
This is also somewhat on hold, since there's no way to add SWC plugins in the most dominant use case being Next.js. FYI @dai-shi

**Reply by axelson** [reply]
@notsidney do you have an example of how you have that setup? It would be great to have a workaround for this.

**Reply by notsidney** [reply]
@axelson Sure, I have a [craco](https://github.com/gsoft-inc/craco) config here: https://github.com/rowyio/rowy/blob/007a4f8387cb88ee7fc9ff90530e880c30aad338/craco.config.js#L15-L44

**Reply by ahungrynoob** [reply]
https://github.com/ahungrynoob/jotai-label-ts-plugin
